beacon_node/store/src/state_cache.rs

```diff
@@ -60,7 +68,8 @@ pub struct HotHDiffBufferCache {
     /// Cache of HDiffBuffers for states *prior* to the `finalized_state`.
     ///
     /// Maps state_root -> (slot, buffer).
-    hdiff_buffers: LruCache<Hash256, (Slot, HDiffBuffer)>,
+    // FIXME(sproul): put OnceCell around HDiffBuffer only?
+    hdiff_buffers: LruCache<Hash256, Arc<OnceCell<(Slot, HDiffBuffer)>>>,
```

Comment:
wdyt about moving the once cell just around the buffer? this would clean up some of the slot-based logic which I think is a bit dodgy now